ops/docker/op-stack-go/Dockerfile.dockerignore

```diff
@@ -2,6 +2,8 @@
 # node_modules, packages (contains git submodules and node modules etc.), devnet dirs etc. all add up a lot.
 *
 
+# keep .git for multicannon
+!/.git
```

Comment:
The alternative is to `git clone` the monorepo during the op-stack-go docker build. Which results in a slightly slower {op-challenger,cannon}-target build.

Reply:
The major downside of this is how often it causes the docker image to rebuild. I suspect we don't want to have to build every old version of cannon each time we build the docker image either.  I wonder if we should just extract the executable from the docker image published as part of the release instead of rebuilding.  We get into some trouble if we need to support new architectures for the docker image though which is a bit tricky. It's probably worth it to avoid slowing down the docker build.

Reply:
Given the list of noignores in this file, pretty much any change to .git means the docker file dependencies are also changed. So the impact on docker build times should be negligible I think.

Though I'm also fine with changing this to extracting the binary from a docker image if you prefer

Reply:
Yeah I think using the existing docker is the way to go, if for no other reason than to avoid doubling the compilation time as soon as we add a second version and avoiding build env differences that may cause us trouble if the base builder image is updated or something like that.

I had forgotten that we don't publish cannon docker images though but it is in op-challenger.  We can pull from the op-challenger images fine but I wonder if we should start releasing cannon specific images - the downside is we'd have to start tagging cannon releases, though we could potentially just do that when we need to capture a legacy version of cannon....

Reply:
Alright. I'll change this.

Note that we have to tag cannon stf changes either way. Already tagged one for this PR - https://github.com/ethereum-optimism/optimism/tree/cannon/v1.0.0-alpha.1. I could have used a commit hash but it's good to create tags whenever we change the STF for easier tracking.

Reply:
Why do we need those tags? If we need to make changes at some point in the future couldn't we just branch off whatever tag we're pulling the docker images from as a starting point?

Reply:
I took a shot at adding a new state version for https://github.com/ethereum-optimism/optimism/pull/12050 in https://github.com/ethereum-optimism/optimism/pull/12050/commits/250973a135dd5b927cacc2a0e67fc7bb189333b4 it's pulling the cannon binary to use from the latest op-challenger release currently.

Reply:
I've removed the .git copy and cloning. The builder now fetches the old cannon executable from the op-challenger v1.1.0 docker image

Reply:
Ah just seeing your previous comment. That looks good. We probably need a better naming scheme for state versions than `getfd`. Since we may make more changes to the VM before holocene.

Reply:
Oh and to answer your question on why we need those tags (getting gh notifications out of order). It's one way to easily locate a version of cannon that supports the `X` state version.
I was thinking of using semvers in the git tags to communicate this. The major semantic version would indicate the changes to the VM STF. And minor versions can be used for regular cannon improvements.

Reply:
Yeah that's a good point actually. Just using the challenger version is too confusing.  Ok that convinces me that we should start publishing cannon docker images that we can use as the source for things.  I'll sort that out. I suspect we only actually need to cut releases for major versions when the STF changes but agree that semantic versioning makes sense.